File: pages/api/NBAPlayerData.py

import sys
from nba_api.stats.endpoints import playercareerstats # type: ignore

def get_total_stat(player_id, stat):
    career_stats = playercareerstats.PlayerCareerStats(player_id=player_id)
    career_stats_data = career_stats.get_data_frames()[0]
    total_stat = career_stats_data[stat].sum()
    return total_stat

# Retrieve command line arguments
player_id = int(sys.argv[1])
stat = sys.argv[2]

# Get the total stat for the player
total_stat = get_total_stat(player_id, stat)
print(total_stat)

File: pages/api/NBAPlayerData.js
//https://nba-stats-db.herokuapp.com/api/

import axios from 'axios';

export async function get_total_stat(player_id, stat) {
  try {
    const response = await axios.get(`https://nba-stats-db.herokuapp.com/api/playerdata/name/${player_id}/?format=json`);
    const career_stats_data = response.data.results[0];
    const total_stat = career_stats_data[stat];
    return total_stat;
  } catch (error) {
    throw new Error('Error fetching player career stats');
  }
}